api/app/Traits/HydratesSnapshot.php

```diff
@@ -19,20 +20,46 @@ trait HydratesSnapshot
      */
     abstract public static function hydrateSnapshot(mixed $snapshot): Model|array;
 
+    protected static function isFieldLocalizedEnum(mixed $snapshot, mixed $snapshotField): bool
+    {
+        // validator for a single localized enum
+        $singleEnumValidator = Validator::make($snapshot, [
+            $snapshotField.'.value' => 'required|string',
+            $snapshotField.'.label.en' => 'required|string',
+            $snapshotField.'.label.fr' => 'required|string',
+        ]);
+        if ($singleEnumValidator->passes()) {
+            return true;
+        }
```

Comment:
This may lead to unexpected behvaiour if we have null localized enum values (idk how likely this is) but:

```php
# Given
'key' => ['value' => null, 'label' => nulll]

# Validator will fail and hydrate to
 'key' => ['value' => null]
 
# instead of
'key' => null
```

Side note is there a reason we are using a validator here? We are really only checking keys so we don't need all the extra stuff that comes with a validator. Though, if we find it easier to reason about validators, we are using them in a similar way in other places so I guess it isn't a huge deal :woman_shrugging: 

```php
// There is probably also a better way to do this :sweat_smile: 
if(is_array($snapshotField) {

  $dot = Arr::dot($snapshotField);
  $expectedKeys = ['value' => '', 'label.en' => '', 'label.fr' => ''];
  
  if(array_is_list($snapshotField) {
     return array_reduce($dot, function ($hasKeys, $field) use($expectedKeys) {
        return $hasKeys && empty(array_diff_key($expectedKeys, $field));
     }, true);
  } else {
     return empty(array_diff_key($expectedKeys, $field));
  }
}
  
return false
```

Reply:
For the second point, yeah, it's readability.  It probably is overkill and a bit slow but I find it much much easier to read and reason about than the code you have above.

For the first point, yeah, let me see what I can do about it.

Reply:
Right, then `en` and `fr` fields could be null, but `label` and `value` can't be.  I've updated the validators and tests to handle that.
[handle null label strings](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11474/commits/0908d468bf5be46f5ece909b72433401ebd48509)